docs/source/development-guidelines.rst

```diff
@@ -0,0 +1,223 @@
+
+   git clone git@github.com:your-github-username/zipline.git
+
+Then check out to a new branch where you can make your changes:
+
+.. code-block:: bash
+		
+   git checkout -b some-short-descriptive-name
+
+The following section assumes you already have virtualenvwrapper and pip installed on your system. If you don't already have them, you'll need some C library dependencies. You can follow the `install guide`__ to get the appropriate dependencies.
+
+__ http://www.zipline.io/install.html 
+
+Suggested installation of Python library dependencies used for development:
+
+.. code-block:: bash
+
+   $ mkvirtualenv Zipline
+   $ ./etc/ordered_pip.sh ./etc/requirements.txt
+   $ pip install -r ./etc/requirements_dev.txt
+   $ pip install -r ./etc/requirements_blaze.txt 
+   $ pip install -r ./etc/requirements_talib.txt 
+
+Finally, you can build the C extensions by running:
+
+.. code-block:: bash
+
+   $ python setup.py build_ext --inplace
+
+To finish, make sure tests pass by running
+
+.. code-block:: bash
+
+   $ nosetests
+
+If you get an error running nosetests after setting up a fresh virtualenv, please try running
+
+.. code-block:: bash
+
+   # where zipline is the name of your virtualenv
+   $ deactivate Zipline
+   $ workon Zipline
+
+
+Development with Docker
+-----------------------
+
+If you want to work with zipline using a `Docker`__ container, you'll need to build the ``Dockerfile`` in the Zipline root directory, and then build ``Dockerfile-dev``. Instructions for building both containers can be found in ``Dockerfile`` and ``Dockerfile-dev``, respectively.
+
+__ https://docs.docker.com/get-started/
+
+
+Style Guide & Running Tests
+---------------------------
+
+We use `flake8`__ for checking style requirements and `nosetests`__ to run Zipline tests. Our `continuous integration`__ tools will run these commands.
+
+__ http://flake8.pycqa.org/en/latest/
+__ http://nose.readthedocs.io/en/latest/
+__ https://en.wikipedia.org/wiki/Continuous_integration
+
+Before submitting patches or pull requests, please ensure that your changes pass when running:
+
+.. code-block:: bash
+
+   $ flake8 Zipline tests
+
+In order to run tests locally, you'll need `TA-lib`__, which you can install on Linux by running:
+
+__ https://mrjbq7.github.io/ta-lib/install.html
+
+.. code-block:: bash
+
+   $ wget http://prdownloads.sourceforge.net/ta-lib/ta-lib-0.4.0-src.tar.gz
+   $ tar -xvzf ta-lib-0.4.0-src.tar.gz
+   $ cd ta-lib/
+   $ ./configure --prefix=/usr
+   $ make
+   $ sudo make install
+
+And for ``TA-lib`` on OS X you can just run:
+
+.. code-block:: bash
+
+   $ brew install ta-lib
+
+You should now be free to run tests:
+
+.. code-block:: bash
+		
+   $ nosetests
+
+
+Continuous Integration
+----------------------
+
+We use `Travis CI`__ for Linux-64 bit builds and `AppVeyor`__ for Windows-64 bit builds.
+
+.. note::
+
+   We do not currently have CI for OSX-64 bit builds, and do not support 32-bit builds.
+
+__ https://travis-ci.org/
```

Comment:
Do you think it would be worth making these link to the zipline project at these sites?